### PR TITLE
Added checks for failed creations of page/procedure

### LINKED ISSUE #145: Create element/page/procedure should first check if server returned 200 before updating UI
If the server does not return a 200 when trying to create a new element/page/procedure, the UI still updates with a new component with id of null. It should instead show a loading spinner and wait for a server response before updating the UI.

### PR DESCRIPTION
**Pull request by Ommy**
When creating a new page or procedure, fails on the save will destroy the object instead. Flash Messages will also be displayed. Resolves #145

### CODE REVIEW

File: src-frontend/app/controllers/procedure.js [hunk 1]
```diff
@@ -19,9 +19,20 @@ export default Ember.Controller.extend({
         return pageModels;
     },
 
+    generateErrorString: function(errors, type) {
+        var message = 'Something went wrong creating the ' + type;
+        errors.forEach(function(error){
+            for (var attr in error.details) {
+                message = attr.toString() + ": " + error.details[attr][0];
```
Comment (Kapin) [inline src-frontend/app/controllers/procedure.js]: shouldn't this just be attr[0] not error.details[attr][0]?

Comment (Kapin) [inline src-frontend/app/controllers/procedure.js]: also single quotes pls

File: src-frontend/app/controllers/procedures/index.js [hunk 2]
```diff
@@ -20,6 +21,15 @@ export default Ember.Controller.extend({
 
             procedure.save().then(function() {
                 procedureController.transitionToRoute('procedure', procedure);
+            }).catch(function(response) {
+                var errorMessage = 'Something went wrong trying to create the procedure';
+                response.errors.forEach(function(error){
+                    for (var attr in error.details) {
+                        errorMessage = attr.toString() + ": " + error.details[attr][0];
```
Comment (Kapin) [inline src-frontend/app/controllers/procedures/index.js]: see above

File: src-frontend/app/controllers/procedures/index.js [hunk 3]
```diff
@@ -20,6 +21,15 @@ export default Ember.Controller.extend({
 
             procedure.save().then(function() {
                 procedureController.transitionToRoute('procedure', procedure);
+            }).catch(function(response) {
+                var errorMessage = 'Something went wrong trying to create the procedure';
```
Comment (Kapin) [inline src-frontend/app/controllers/procedures/index.js]: this should call generateErrorString

File: src-frontend/app/controllers/procedure.js (hunk 1 above)
Comment (Ommy) [inline src-frontend/app/controllers/procedure.js]: No, cause attr will be the attribute such as `display_index`, so we access the value of that message like if it was `display_index: "Can't be blank!"`

Comment (Kapin) [inline src-frontend/app/controllers/procedure.js]: :skull: :gun:

File: src-frontend/app/controllers/procedures/index.js (hunk 3 above)
Comment (Ommy) [inline src-frontend/app/controllers/procedures/index.js]: I can't call that function from within this controller. @ConnorCimowsky is there some Ember convention for putting JS helper functions (not for Handlebars)

File: src-frontend/app/controllers/procedure.js (hunk 1 above)
Comment (Trinovantes) [inline src-frontend/app/controllers/procedure.js]: It should be double quotes for UI strings. We only use single quotes for key-like strings

File: src-frontend/app/controllers/procedure.js [hunk 4]
```diff
@@ -54,6 +65,10 @@ export default Ember.Controller.extend({
                 newElement.save().then(function() {
                     procedureController.send('editPage', newPage);
                 });
+            }).catch(function(response) {
+                var errorMessage = procedureController.generateErrorString(response.errors, 'page');
+                flashMessages.danger(errorMessage);
+                newPage.destroy();
```
Comment (Trinovantes) [inline src-frontend/app/controllers/procedure.js]: does this route us back to the state before the page is created?

**Comment by Trinovantes** [comment]
Stopped development with Ember

**Pull request closed without merging**